server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeFilters.java

```diff
@@ -88,6 +89,34 @@ private boolean matchByIP(String[] values, @Nullable String hostIp, @Nullable St
         return false;
     }
 
+    /**
+     * Removes any filters that should not be considered, returning a new
+     * {@link DiscoveryNodeFilters} object. If the filtered object has no
+     * filters after trimming, {@code null} is returned.
+     */
+    @Nullable
+    public static DiscoveryNodeFilters trim(@Nullable DiscoveryNodeFilters original) {
+        if (original == null) {
+            return null;
+        }
+
+        Map<String, String[]> newFilters = original.filters.entrySet().stream()
+            // Remove all entries that start with "_tier", as these will be handled elsewhere
+            .filter(entry -> {
+                String attr = entry.getKey();
+                if (attr != null && attr.startsWith("_tier")) {
+                }
```

Comment:
oops, I think this is a leftover?

Reply:
whoops yep, leftover from debugging :X I'll remove